src/pushsource/_impl/model/vms.py

```diff
@@ -54,6 +54,10 @@ class VMIRelease(object):
     type = attr.ib(type=str, default=None, validator=optional_str)
     """Release type, typically "ga" or "beta"."""
 
+    marketplace_title = attr.ib(type=str, default=None, validator=optional_str)
+    """Customer facing title or title template for the marketplace release. For example,
```

Comment:
Is there something which can be said more specific than "template"?  Like, "jinja-compatible template" if that's what it is?

Reply:
... or: template intended for usage with `str.format`.  But it kind of seems like the variables need to be defined somewhere.  Isn't it going to crash if you try to render the format without providing all of the variables it uses?

The JIRA issue also doesn't explain anything about it.  I don't like the idea of exposing some kind of template of an undefined format, a year or two down the line we could be arguing about which component is buggy if the template produced by one tool doesn't meet the format expected by some other tool.

Reply:
Yet another way is to use some regex to delimit our "templates", like:

"This will be the version @VERSION@ of product @PRODUCT@" and we do the formatting using some kind of regex/replace.

Reply:
But supposing we agree that the accepted variables may be any attribute of `VMIRelease` / `AmiRelease`.

The following snippet could be a way to format the string by any tooling:

```
string_to_format = push_item.marketplace_title
keywords = re.findall(r'{\w+}', string_to_format)
params = {}
for k in keywords:
    kw = k.replace("{", "").replace("}", "")
    params[kw] = getattr(pushitem.release, kw, "")

return string_to_format.format(**params)
```

WDYT?

Reply:
According to a quick test I did just now, str.format will happily ignore any unused keywords without generating a warning or error.  So could the above code be simplified more to this? (Assuming the caller is willing to call a method from `attr` library)

```
push_item.marketplace_title.format(**attr.asdict(pushitem.release))
```

But going that far, I start to wonder if this attribute should rather be named `marketplace_title_template` and then `marketplace_title` is a `@property` which more or less contains the above code?

Anyway, I like the following two options, documentation-wise:

> The template is of the form used by `str.format`, with available keywords being all of the documented fields on `VMIRelease` and `AMIRelease` classes.

Or:

> The template is of the form used by (some reference/link to some other project)

For the latter approach I wouldn't care much about the details, just that "something" somewhere is responsible for defining the format, so that we know it's not the responsibility of pushsource library.

Reply:
@rohanpm I've changed it completely to follow your suggestion:

- Use the `marketplace_title_template` in `VMIPushItem` to receive any variables from `VMIRelease`
- Create the property `marketplace_title` in `VMIRelease` to format the template
- Updated tests

Please let me know if it's good now